autoprotocol/instruction.py

```diff
@@ -86,6 +100,119 @@ def filter_criteria(item):
             ]
         return data
 
+    def _set_informatics(self):
```

Comment:
as per the previous review, I would prefer if we can pass an instantiated Informatics-class directly to the instruction-helper. I.e. `p.provision(..., informatics=[AttachCompounds(...)])`. 
Otherwise, if we do not want to end up manually adding defining that additional option for all Instructions, we can consider the `append` pattern, where you attach an Informatics to the Instruction in a separate call.  I.e. `p.provision(...); p.attach_informatics(AttachCompounds(...), instruction_index=p.get_instruction_index())`

Thinking more about it, I'm increasingly favoring the latter. Something that we can certainly get more user feedback on though.

Reply:
Hmm. I don't mind `p.attach_informatics` route either, but we still need to associate `AttachCompounds` to a specific instruction. Should we have instruction index as a user input? Or assume `attach_informatics` to have effect on the previous instruction in the sequence of events?

Reply:
oh, never mind. You did mention instruction_index. :p 
Ya, I initially thought of that (in https://github.com/strateos/autoprotocol_standard_changes/pull/16), but got turned down since it's harder to deserialize...